03/product.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+
+export default class Product extends React.Component {
+    
+    render() {
+        const {d, btn, func, id} = this.props;
+
+        const insertButton = () => {
+            if (btn === "cart") {
+                return (
+                    <button onClick={func}>Usuń z koszyka</button>
+                )
+            }
+            if (btn === "category") {
+                return (
+                    <button class={`cl` + d.id} onClick={func}>Dodaj do koszyka</button>
+                )
+            }
+            return
+        }
```

Comment:
Niepotrzebnie utrudniasz sobie życie.
Wystarczy: `<button onClick={e => this.props.handleClick(this.props.id)}>{this.props.text}</button>`.
To w `props` przekazujesz odpowiednie dane i nie trzeba robić żadnych dodatkowych warunków.